### PR TITLE
`packages/modeldb-sqlite-wasm` - implement modeldb on OPFS sqlite 

### PR DESCRIPTION
**Pull request by rjwebb**
This PR adds an implementation of ModelDB that uses Sqlite compiled to WASM running inside a worker thread with the .sqlite3 file stored in OPFS. This means that ModelDB can be implemented in the browser using a SQL relational database instead of IndexedDB.

Communication between the browser thread and the worker thread is handled by the [comlink](https://github.com/GoogleChromeLabs/comlink) library, this lets us call functions inside the worker thread instead of having to think in terms of events and messages. It also lets us pass "proxy" functions into the worker thread for doing subscriptions for `useLiveQuery`.

## Changes

Add a new module `packages/modeldb-sqlite-wasm`, a ModelDB implementation built on [sqlite-wasm](https://www.npmjs.com/package/@sqlite.org/sqlite-wasm) for the browser. To create a ModelDB instance, call the `ModelDB.initialize` static method, which takes an object with two arguments: "models" (the model schema, same as with the other implementations) and "path" (an optional string). If the "path" argument is given, then a database persisted on OPFS will be created at the given path, if it is not given then a transient in-memory database will be created. To use OPFS, the server must be set up so that the `Cross-Origin-Opener-Policy` header is set to `same-origin` and the `Cross-Origin-Embedder-Policy` header is set to `require-corp` in the response headers. In practice this means that we can't use OPFS on web hosts like GitHub Pages where we can't configure the response headers.

Update the browser target in `@canvas-js/core` so that if a "path" init value is set, Canvas will use Sqlite + Wasm + OPFS for the Canvas database and Gossiplog, otherwise it will `@canvas-js/modeldb-idb` (the current default).

Update the Chat example app so that it uses the Sqlite Wasm ModelDB with OPFS. The vite config has also been updated to serve the required headers.

Update the ModelDB tests so that they can also run in Puppeteer, this lets us test ModelDB implementations that can only run in the browser. We do this by running a test server which exposes all of the imported functions/classes used by the test as globals. Then when we run the tests, we can stringify the test functions and run them inside the page using `page.evaluate(function)`.

Update CI to run the test server alongside the ModelDB tests (these are currently running in CI).

## How has this been tested?

- [x] CI tests pass
- [x] Tested with example-chat (including login, all signers, and exchanging messages)
- [ ] Tested with `@canvas-js/test-network`: (optional)

## Does this contain any breaking changes to external interfaces?

- [ ] Contract interfaces
- [ ] Core interface
- [ ] CLI
- [ ] Data storage formats, including IndexedDB, SQLite, or filesystem storage (will this break existing apps?)

### CODE REVIEW

File: packages/modeldb-sqlite-wasm/src/TransientModelDB.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	Effect,
+	ModelValue,
+	ModelSchema,
+	QueryParams,
+	Config,
+	getFilter,
+} from "@canvas-js/modeldb"
+import { InnerModelDB } from "./InnerModelDB.js"
+import { Awaitable } from "@canvas-js/interfaces"
+import assert from "assert"
```
Comment (joeltg) [inline packages/modeldb-sqlite-wasm/src/TransientModelDB.ts:15]: `assert` is a NodeJS package and we don't want to make people polyfill things if necessary. we export our own as `import { assert } from "@canvas-js/utils"`

Comment (rjwebb) [reply packages/modeldb-sqlite-wasm/src/TransientModelDB.ts:15]: Good catch

File: packages/modeldb/test/server/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+		"moduleDetection": "force",
+		"noEmit": true,
+
+		"strict": true,
+
+		"noUnusedParameters": true,
+		"noFallthroughCasesInSwitch": true
+	},
+	"include": ["src"],
+	"references": [{ "path": "../../../modeldb-sqlite-wasm" }]
```
Comment (joeltg) [inline packages/modeldb/test/server/tsconfig.json:22]: add reference to `../../.` too (@canvas-js/modeldb)

File: packages/modeldb/package.json [hunk 3]
```diff
@@ -13,11 +13,15 @@
 	"types": "./lib/index.d.ts",
 	"sideEffects": false,
 	"scripts": {
+		"test-server": "cd test/server && vite",
 		"test": "ava --serial"
 	},
 	"devDependencies": {
+		"@canvas-js/modeldb-sqlite-wasm": "^0.10.3",
```
Comment (joeltg) [inline packages/modeldb/package.json:20]: local packages don't actually need to be added to devDependencies

**Review by joeltg — APPROVED** [review]

**Pull request merged**